### PR TITLE
p4: init at 2020.1.1991450

### PR DESCRIPTION
**Pull request by corngood**
###### Motivation for this change

Ad the binary blob perforce command line client.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/p4/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  version = "2020.1.1991450";
+
+  src = fetchurl {
+    url = "https://cdist2.perforce.com/perforce/r20.1/bin.linux26x86_64/helix-core-server.tgz";
+    sha256 = "0nhcxhwx3scx6vf7i2bc8j0b1l57lmq9bfy1cfbfbqasd3an721k";
+  };
+
+  # Work around the "unpacker appears to have produced no directories"
+  # case that happens when the archive doesn't have a subdirectory.
+  setSourceRoot = "sourceRoot=`pwd`";
```
Comment (veprbl) [inline pkgs/applications/version-management/p4/default.nix:14]: How about
```nix
sourceRoot = ".";
```
?

File: pkgs/applications/version-management/p4/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    for f in $out/bin/*; do
+      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $f
+    done
+  '';
+
+  meta = {
+    description = "Perforce Command-Line Client";
+    homepage = "https://www.perforce.com";
+    license = lib.licenses.unfreeRedistributable;
```
Comment (veprbl) [inline pkgs/applications/version-management/p4/default.nix:30]: Maybe you can prove me wrong, but it looks to me like this is `unfree`:
```suggestion
    license = lib.licenses.unfree;
```

File: pkgs/applications/version-management/p4/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp p4 p4broker p4d p4p $out/bin
+
+    for f in $out/bin/*; do
+      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $f
+    done
```
Comment (veprbl) [inline pkgs/applications/version-management/p4/default.nix:24]: Can you use `autoPatchelfHook` instead?

File: pkgs/applications/version-management/p4/default.nix (hunk 2 above)
Comment (corngood) [reply pkgs/applications/version-management/p4/default.nix:30]: This was copied from `p4v`.  Unfortunately all I can find are licenses for dependencies (e.g. boost), which are all free.  I'll change it to `unfree`.

Perhaps we should consider changing p4v as well?

Comment (veprbl) [reply pkgs/applications/version-management/p4/default.nix]: I also only found licenses for the dependencies, but downloading from the official site seems to require agreeing with EULA[1]. I would think the change to `p4v.meta.license` is warranted as well.

[1] https://www.perforce.com/terms-use#eula

**Review by veprbl — APPROVED** [review]
Diff LGTM

**Pull request merged**